tests/test_funcs.py

```diff
@@ -28,91 +28,96 @@ class TestAtLeastND:
     def test_0D(self):
         x = xp.asarray(1)
 
-        y = atleast_nd(x, ndim=0, xp=xp)
+        y = atleast_nd(x, ndim=0)
         assert_array_equal(y, x)
 
-        y = atleast_nd(x, ndim=1, xp=xp)
+        y = atleast_nd(x, ndim=1)
         assert_array_equal(y, xp.ones((1,)))
 
-        y = atleast_nd(x, ndim=5, xp=xp)
+        y = atleast_nd(x, ndim=5)
         assert_array_equal(y, xp.ones((1, 1, 1, 1, 1)))
 
     def test_1D(self):
         x = xp.asarray([0, 1])
 
-        y = atleast_nd(x, ndim=0, xp=xp)
+        y = atleast_nd(x, ndim=0)
         assert_array_equal(y, x)
 
-        y = atleast_nd(x, ndim=1, xp=xp)
+        y = atleast_nd(x, ndim=1)
         assert_array_equal(y, x)
 
-        y = atleast_nd(x, ndim=2, xp=xp)
+        y = atleast_nd(x, ndim=2)
         assert_array_equal(y, xp.asarray([[0, 1]]))
 
-        y = atleast_nd(x, ndim=5, xp=xp)
+        y = atleast_nd(x, ndim=5)
         assert_array_equal(y, xp.reshape(xp.arange(2), (1, 1, 1, 1, 2)))
 
     def test_2D(self):
         x = xp.asarray([[3]])
 
-        y = atleast_nd(x, ndim=0, xp=xp)
+        y = atleast_nd(x, ndim=0)
         assert_array_equal(y, x)
 
-        y = atleast_nd(x, ndim=2, xp=xp)
+        y = atleast_nd(x, ndim=2)
         assert_array_equal(y, x)
 
-        y = atleast_nd(x, ndim=3, xp=xp)
+        y = atleast_nd(x, ndim=3)
         assert_array_equal(y, 3 * xp.ones((1, 1, 1)))
 
-        y = atleast_nd(x, ndim=5, xp=xp)
+        y = atleast_nd(x, ndim=5)
         assert_array_equal(y, 3 * xp.ones((1, 1, 1, 1, 1)))
 
     def test_5D(self):
         x = xp.ones((1, 1, 1, 1, 1))
 
-        y = atleast_nd(x, ndim=0, xp=xp)
+        y = atleast_nd(x, ndim=0)
         assert_array_equal(y, x)
 
-        y = atleast_nd(x, ndim=4, xp=xp)
+        y = atleast_nd(x, ndim=4)
         assert_array_equal(y, x)
 
-        y = atleast_nd(x, ndim=5, xp=xp)
+        y = atleast_nd(x, ndim=5)
         assert_array_equal(y, x)
 
-        y = atleast_nd(x, ndim=6, xp=xp)
+        y = atleast_nd(x, ndim=6)
         assert_array_equal(y, xp.ones((1, 1, 1, 1, 1, 1)))
 
-        y = atleast_nd(x, ndim=9, xp=xp)
+        y = atleast_nd(x, ndim=9)
         assert_array_equal(y, xp.ones((1, 1, 1, 1, 1, 1, 1, 1, 1)))
 
     def test_device(self):
         device = xp.Device("device1")
         x = xp.asarray([1, 2, 3], device=device)
-        assert atleast_nd(x, ndim=2, xp=xp).device == device
+        assert atleast_nd(x, ndim=2).device == device
+
+    def test_xp(self):
+        x = xp.asarray(1)
+        y = atleast_nd(x, ndim=0, xp=xp)
+        assert_array_equal(y, x)
```

Comment:
(here and in all other `test_xp`) I couldn't come up with an interesting use case where an explicit xp yields different output from xp=None?

Reply:
Yep, it shouldn't affect output, since this library assumes inputs are already arrays.